src/bootctl/bootctl-status.c

```diff
@@ -525,10 +527,13 @@ int verb_status(int argc, char *argv[], void *userdata) {
                         (void) efi_get_variable_string_and_warn(EFI_LOADER_VARIABLE_STR("StubDeviceURL"), &stub_url);
 
                         if (!sd_id128_is_null(stub_partition_uuid)) {
-                                if (!(!sd_id128_is_null(esp_uuid) && sd_id128_equal(esp_uuid, stub_partition_uuid)) &&
-                                    !(!sd_id128_is_null(xbootldr_uuid) && sd_id128_equal(xbootldr_uuid, stub_partition_uuid)))
-                                        printf("WARNING: The stub loader reports a different UUID than the detected ESP or XBOOTDLR partition ("SD_ID128_UUID_FORMAT_STR" vs. "SD_ID128_UUID_FORMAT_STR"/"SD_ID128_UUID_FORMAT_STR")!\n",
-                                               SD_ID128_FORMAT_VAL(stub_partition_uuid), SD_ID128_FORMAT_VAL(esp_uuid), SD_ID128_FORMAT_VAL(xbootldr_uuid));
+                                if (!sd_id128_equal(esp_uuid, stub_partition_uuid) &&
+                                    !sd_id128_equal(xbootldr_uuid, stub_partition_uuid))
```

Comment:
hmm, the check here was simply broken.

the idea was to only print this warning in case esp_uuid/xbootldr_uuid are actually known. the loader check further up got that right, this one is fucked up though.

Reply:
Indeed. Reworked. PTAL.